sys-apps/edac-utils/edac-utils-0.18-r1.ebuild

```diff
@@ -1,27 +1,41 @@
 # Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=7
 
-inherit eutils ltprune
+inherit autotools flag-o-matic
 
 DESCRIPTION="Userspace helper for Linux kernel EDAC drivers"
 HOMEPAGE="https://github.com/grondo/edac-utils"
 SRC_URI="https://github.com/grondo/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="amd64"
+KEYWORDS="~amd64"
 IUSE="debug"
 
 DEPEND="sys-fs/sysfsutils"
 RDEPEND="${DEPEND}
 	sys-apps/dmidecode"
 
 src_prepare() {
+	default
+
+	# <sys-devel/gcc-10, <sys-devel/binutils-2.34 do not support LDPT_ADD_SYMBOLS_V2
+	# https://sourceware.org/bugzilla/show_bug.cgi?id=25355
+	local gcc_vers=$(gcc-version)
+	local ld_vers=$(ld.bfd -v | awk '{ print $4 }')
+
+	if tc-is-gcc && ver_test ${gcc_vers} -lt 10 ||
```

Comment:
A comment here explaining this would help, but maybe we should also put some of the tests into their own variables, like:
local gcc_too_old=....

Maybe I should ask some of the toolchain folks if this could be simplified.

Reply:
Noted on adding comments, but I'm not too sure what you mean after that.

It feels somewhat heavy already for only checking versions. Perhaps a case statement would be cleaner and easier to work with?